biogrid/libs/biogrid-simulator/src/lib/community/ruralarea.ts

```diff
@@ -0,0 +1,48 @@
+import { Towns } from '@biogrid/grid-simulator';
+import { Building } from '../building';
+
+/**
+ * A rural area that represents a community where a microgrid can function.
+ */
+export class RuralArea implements Towns{
+
+  /** A town of buildings that use energy. */
+  private town: Building[];
+
+  /**
+   * @param {Building[]} town A given town.
+   */
+  constructor(town: Building[]) {
+    this.town = town;
+  }
+
+  /**
+   * Gets our current town.
+   * @return {Building[]} Returns our town.
+   */
+  getTown(){
+    return this.town;
+  }
+
+  /**
+   * This method gets a building in a town.
+   * @return {Building} Returns the building if found in the list, null if not.
+   */
+  getBuilding(building: Building){
```

Comment:
I'm not really clear on how this is a useful method. It seems like you are passing in a building already, and returning that same building. Maybe getBuildingById or hasBuilding or something similar would be helpful, but since you are passing in a full building object and returning the same building object, I'm not sure this function is helpful.

Reply:
Ah yes you're right. I'll try a new way of doing this method.

Reply:
@aosmann7  try to implement it with something like passing coordinates or position for the housing or id, it can also be something like zip code